firebase/functions/src/post-lab-write.spec.ts

```diff
@@ -0,0 +1,102 @@
+import 'jest';
+import * as sinon from 'sinon';
+import * as admin from 'firebase-admin';
+import * as setupOfflineTests from 'firebase-functions-test';
+
+const test = setupOfflineTests();
+
+test.mockConfig({
+  firebase: {},
+  fb_service_account: {}
+});
+
+describe('Firebase Functions', () => {
+  let postLabWriteFn, hashDirectoryFn, adminInitStub, oldDatabase;
+
+  describe('postLabWrite', () => {
+    beforeEach(() => {
+      adminInitStub = sinon.stub(admin, 'initializeApp');
+      oldDatabase = admin.database;
+      postLabWriteFn = require('./post-lab-write').postLabWrite;
+      hashDirectoryFn = require('./post-lab-write').hashDirectory;
```

Comment:
This can be just imported at the top as it doesn't require any stubbing. It can be tested as a normal unit test.

Not sure if `postLabWrite` can be imported at the top as well, I don't see why not but might be missing something.

Reply:
According to the firebase unit testing docs it's important to import the functions under test **after** the `admin` has been initialized/stubbed. The reason for this is because usually you call `admin.initializedApp()` within the file that exports all firebase functions (the file you'd `require()` here). I'm not requiring the `index.ts` here because I wanted to get something working for a single function for now. This might change as I'm upgrading all other functions to `1.0` as well, and then this require statement needs to happen after the stub of `admin.initializeApp()`.

Reply:
More on that here: https://firebase.google.com/docs/functions/unit-testing#importing_your_functions